src/reader/mod.rs

```diff
@@ -127,6 +127,36 @@ macro_rules! configure_methods {
     };
 }
 
+/// Generalization of `read_to_end` method for buffered and borrowed readers
+macro_rules! read_to_end {
```

Comment:
I agree in theory with deduplication, but it would be better to wait until everything is settled before doing it.  Please drop this commit for now, we can bring it back later if it still makes sense.

Reply:
Removed